### DISCUSSION TITLE
복수 API 설정 中 XML 타입 네트워크 응답 처리 관련 토론

### DISCUSSION THREAD

**Discussion by sunjoolee** [WDTT-CodingClub/NBDream#73] [General]
## Issue
#70 

## Discussion
지금 네트워크 응답을 String 타입으로 받아온 다음, 응답이 Json 타입으로 컨버트하고 네트워크 DTO에 맞게 파싱하는 방식이죠?

네트워크 요청 오류 시, 아래와 같은 형식으로 응답이 오는데
```xml
<OpenAPI_ServiceResponse>
        <cmmMsgHeader>
                <errMsg>SERVICE ERROR</errMsg>
                <returnAuthMsg>SERVICE_KEY_IS_NOT_REGISTERED_ERROR</returnAuthMsg>
                <returnReasonCode>30</returnReasonCode>
        </cmmMsgHeader>
</OpenAPI_ServiceResponse>
```
이 응답은 XML 타입이라 Json 타입으로 컨버팅이 안되고 throwable을 날리네요
```
kotlinx.serialization.json.internal.JsonDecodingException: Unexpected JSON token at offset 0: Expected start of the object '{', but had '<' instead at path: $
JSON input: <?xml version="1.0" encoding=".....
```
공공 데이터 포털에서 제공하는 API 대부분이 네트워트 오류 응답을 XML로 주는 걸로 알고 있어서
XML 응답도 파싱할 수 있도록 구현을 추가해야 할까요?

**Comment by sunjoolee** [comment]
농사로 농작업일정 Json 조회 API 응답이 성공해도 XML 형식으로 받아와지네요! 
XML 파싱을 필수적으로 구현해야할 것 같습니당 😢
아니면 농사로 API만 XML 컨버터 라이브러리(tikxml) 사용하는 Retrofit Object 따로 만들어서 사용할까요?

**Reply by Guri999** [reply]
tikxml라이브러리가 어떤식으로 파싱하는지 봐야할것 같습니다

**Comment by Guri999** [comment]
아 네 json 전용이라 xml은 따로 만들어야 할것 같아요

Xml 형식 파싱하는건 제가 잘 몰라서,,

공공데이터에서 데이터 타입을 설정 할 수 없다면 따로 만들어야 할듯 합니다
